contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -223,28 +223,26 @@ contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions
    * @inheritdoc IFunctionsRouter
    */
   function fulfill(
-    bytes32 requestId,
     bytes memory response,
     bytes memory err,
     uint96 juelsPerGas,
     uint96 costWithoutFulfillment,
-    address transmitter
+    address transmitter,
+    IFunctionsRequest.Commitment memory commitment
   ) external override returns (uint8 resultCode, uint96 callbackGasCostJuels) {
     _whenNotPaused();
 
-    Commitment memory commitment = s_requestCommitments[requestId];
+    if (keccak256(abi.encode(commitment)) != s_requestCommitments[commitment.requestId]) {
+      resultCode = 2; // FulfillResult.INVALID_REQUEST_ID
+      return (resultCode, callbackGasCostJuels);
+    }
 
     if (msg.sender != commitment.coordinator) {
       revert OnlyCallableFromCoordinator();
     }
 
-    if (commitment.client == address(0)) {
-      resultCode = 2; // FulfillResult.INVALID_REQUEST_ID
-      return (resultCode, callbackGasCostJuels);
-    }
-
     // Check that the transmitter has supplied enough gas for the callback to succeed
-    if (gasleft() < commitment.callbackGasLimit + commitment.gasAfterPaymentCalculation) {
+    if (gasleft() < commitment.callbackGasLimit + commitment.gasOverheadAfterCallback) {
```

Comment:
?

Reply:
It's the same, just using a consistent name from the Commitment interface

Reply:
Yep.  @bolekk see my comment: https://github.com/smartcontractkit/chainlink/pull/9898#discussion_r1281108875